src/utils.c

```diff
@@ -0,0 +1,54 @@
+#include "data.table.h"
+#include <Rdefines.h>
+
+void any_na_integer(int *x, uint64_t nx, bool *ans, uint64_t *first, bool parallel) {
+  bool found=false;
+  #pragma omp parallel for if (parallel) num_threads(getDTthreads())
+  for (uint64_t i=0; i<nx; i++) {
+    if (found) continue;
+    if (x[i]==NA_INTEGER) {
+      found = true;
+      first[0] = i;
```

Comment:
If multiple threads find an NA simultaneously and try to write into `*first`, an Undefined Behavior will occur (possibly with the result that `*first` will end up containing some garbage value).

A bigger problem though if some thread finds an NA value, but it is not the FIRST value in the vector. Say if each thread gets to check 1000 rows, and there are NA values in rows 999 and 1000, then the second thread will find the NA in row 1000 immediately and signal all threads to stop, long before the first thread gets to row 999.

A better strategy would be something like this:
```
found_index = INTMAX;
#omp for dynamic(1) reduce(min:found_index)
for( i = 1:NBATCHES ) {
  if (found) continue;
  for (row = i*BATCH_SIZE:(i+1)*BATCH_SIZE) {
    if (x[row] is na) {
      found = true;
      found_index = row; 
      break;
    }
  }
}
```

Reply:
I removed `first` related logic for now. Originally added it so extra computation is not lost, but it is not yet used. Your code is put there in comment so when needed can be added.